tensorflow-framework/src/main/java/org/tensorflow/framework/initializers/Ones.java

```diff
@@ -53,13 +55,22 @@ public Ones(Ops tf) {
     super(tf);
   }
 
-  /** {@inheritDoc} */
+  /**
+   * Generates the operation used to perform the initialization.
+   *
+   * @param dims the shape dimensions
+   * @param type the data type of tensor
+   * @param <U> The data Type for initializer operation
+   * @return An operand for the initialization.
+   * @throws IllegalArgumentException if the data type is not a TNumber or TBool
+   */
   @Override
-  public <T extends TType> Operand<T> call(Operand<TInt64> dims, Class<T> type) {
+  public <U extends TType> Operand<U> call(Operand<TInt64> dims, Class<U> type) {
     if (!TNumber.class.isAssignableFrom(type) && type != TBool.class) {
       throw new IllegalArgumentException(
           "Tensor type must be numeric or boolean: " + type.getSimpleName());
     }
-    return tf.fill(dims, tf.dtypes.cast(tf.constant(1.0), type));
+
+    return cast(tf, tf.fill(dims, tf.constant(1)), type);
```

Comment:
This switched the order of the fill and the cast. Is there a reason to prefer one way over the other?

Reply:
I am not sure if it make a difference, but might be faster to fill with the right type first. I will fix.